src/Campaigns/Migrations/Tables/CreateCampaignsTable.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Give\Campaigns\Migrations\Tables;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+/**
+ * @unreleased
+ * Creates give_campaigns table
+ */
+class CreateCampaignsTable extends Migration
+{
+    /**
+     * @inheritdoc
+     */
+    public static function id(): string
+    {
+        return 'give-campaigns-create-give-core-campaigns-table';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function title(): string
+    {
+        return 'Create give_campaigns table from core';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function timestamp(): string
+    {
+        return strtotime('2024-08-26 00:00:00');
```

Comment:
I think it doesn't matter, but yeah, we can do that.

Reply:
The timestamp determines run order, if I understand correctly.

It may not be an issue in this particular instance, but I think it would be prudent to maintain the timestamp as a general rule (not that we have ever moved a migration like this).

Reply:
> The timestamp determines run order, if I understand correctly.

Yes, and we want this migration to run later than the migration from P2P (if P2P is already installed). This one will add the `campaign_type` column if the give_campaign table exists. Also, we will have to add a check (give version probably) in P2P to conditionally run P2P migration for creating the campaign table.

Reply:
> This one will add the campaign_type column if the give_campaign table exists.

Good point. Should we introduce these changes as an additional migration?

You mentioned previously that `dbDelta` can handle this automatically, but we haven't used it that way. Rather, we've used separation migrations that run synchronously. Perhaps we should port the "add campaign" migration and in addition introduce a new migration that modifies the original table.

Reply:
> we will have to add a check (give version probably) in P2P to conditionally run P2P migration for creating the campaign table.

We won't necessarily have to add a logic check, if we leverage the migration and timestamps.

If the previous migration has run via P2P, then porting the migration and the timestamp should mean that it would inherently not run again.

Reply:
> Good point. Should we introduce these changes as an additional migration? You mentioned previously that dbDelta can handle this automatically, but we haven't used it that way. Rather, we've used separation migrations that run synchronously. Perhaps we should port the "add campaign" migration and in addition introduce a new migration that modifies the original table.

I think I know how you feel about using `dbDelta`, I'm a little anxious about using that as well 😅  
But `dbDelta` does the job. I'm perfectly fine going with the "standard" way of doing things as it adds a bit more clarity to what is going on. So yeah, let's add column type as a separate migration.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7519/commits/986cfe14895a177431734823a8b4d121549cb676